sdk/python/lib/pulumi/automation/_stack.py

```diff
@@ -299,7 +311,9 @@ def on_exit_fn():
         try:
             up_result = self._run_pulumi_cmd_sync(args, on_output)
             outputs = self.outputs()
-            summary = self.info(show_secrets)
+            # If it's a remote workspace, explicitly set show_secrets to False to prevent attempting to
+            # load the project file.
+            summary = self.info(show_secrets and not self._is_remote)
             assert summary is not None
```

Comment:
I know this isn't part of the PR, but should we really have asserts in production code?

Reply:
As long as we are asserting on our own logic (not user input), I think its fine. A `Exception` is better than misunderstanding what is going on and potentially doing the wrong thing.